src/corehost/common/pal.unix.cpp

```diff
@@ -57,6 +57,37 @@ bool pal::touch_file(const pal::string_t& path)
     return true;
 }
 
+void* pal::map_file_readonly(const pal::string_t& path, size_t& length)
+{
+	int fd = open(path.c_str(), O_RDONLY, (S_IRUSR | S_IRGRP | S_IROTH));
+	if (fd == -1)
+	{
+		trace::warning(_X("Failed to map file. open(%s) failed with error %d"), path.c_str(), errno);
+		return nullptr;
+	}
+
+	struct stat buf;
+	if (fstat(fd, &buf) != 0)
+	{
+		trace::warning(_X("Failed to map file. fstat(%s) failed with error %d"), path.c_str(), errno);
+        close(fd);
+        return nullptr;
+	}
+
+	length = buf.st_size;
+	void* address = mmap(nullptr, , PROT_READ, MAP_SHARED, fd, 0);
```

Comment:
You're probably missing the `length` parameter. This probably won't even compile.

Reply:
Sorry, I'll fix this.
I carved out this branch from a larger change that attempts to load IL from the bundle. But I missed a fix that was out of sequence.